compiler/testData/diagnostics/tests/enum/enumSubjectTypeCheck.fir.kt

```diff
@@ -5,7 +5,7 @@ enum class En { A, B, С }
 fun foo() {
     // nullable variable
     val en2: Any? = En.A
-    if (en2 is En) {
+    if (<!USELESS_IS_CHECK!>en2 is En<!>) {
```

Comment:
Unlike cast (`as`),  FE1.0's `is` check doesn't see if the use of type checking defines the expression type. I.e., it never checks exact cast, but always up cast. It's unclear whether FE1.0 doesn't have smartcast info at this point or intentionally uses the user-specified type (`Any?`). In contrast, FIR has smartcast info here (`En`) from the initial value, and regards this (and below) `is` is redundant.